### PR TITLE
Rolluplist component

### PR DESCRIPTION
**Pull request by corwintines**
## Description
- Break out rollup information into separate data files. This will function as a source of truth for L2 data as we move forward into other L2 pages. 
- Rollup component to display website, devdocs, and l2beat info for each rollup in devdocs

https://ethereumorgwebsitedev01-rolluplistcomponent.gtsb.io/en/developers/docs/scaling/optimistic-rollups/
https://ethereumorgwebsitedev01-rolluplistcomponent.gtsb.io/en/developers/docs/scaling/zk-rollups/

### CODE REVIEW

File: src/components/RollupProductDevDoc.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                </H3>
+                <div>
+                  <Link to={rollup.website}>Website</Link>
+                </div>
+                <div>
+                  <Link to={rollup.developerDocs}>Developer docs</Link>
+                </div>
+                <div>
+                  <Link to={rollup.l2beat}>L2BEAT</Link>
+                </div>
```
Comment (minimalsm) [inline src/components/RollupProductDevDoc.js:54]:
```suggestion
                <ul>
                 <li>
                    <Link to={rollup.website}>Website</Link>
                  </li>
                  <li>
                    <Link to={rollup.developerDocs}>Developer docs</Link>
                  </li>
                  <li>
                    <Link to={rollup.l2beat}>L2BEAT</Link>
                  </li>
                </ul>
```

File: src/components/RollupProductDevDoc.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    <div>
+      {data[rollupType].map((rollup) => {
+        return (
+          <ProductCard>
+            <Content>
+              <div>
+                <H3>
+                  {rollup.name}
+                  {rollup.note.length > 0 && ` (${rollup.note})`}
+                </H3>
```
Comment (minimalsm) [inline src/components/RollupProductDevDoc.js:45]: I think this should be an h4 as it's a child of an h3.

Screenshot 2022-02-28 at 12 35 23


```suggestion
                <H4>
                  {rollup.name}
                  {rollup.note.length > 0 && ` (${rollup.note})`}
                </H4>
```

File: src/components/RollupProductDevDoc.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+// Components
+import Link from "./Link"
+
+// Data
+import optimisticRollupList from "../data/layer-2-data-files/optimistic-rollups.json"
+import zkRollupList from "../data/layer-2-data-files/zk-rollups.json"
+
+// Styles
+const H3 = styled.h3`
```
Comment (minimalsm) [inline src/components/RollupProductDevDoc.js:13]:
```suggestion
const H4 = styled.h4`
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
Nice job pulling this out into data files. Left a few comments below.

Not sure if it's deliberate but the cards have a white background on lightmode.

Screenshot 2022-02-28 at 12 37 21

File: src/components/RollupProductDevDoc.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                  {rollup.note.length > 0 && ` (${rollup.note})`}
+                </H3>
+                <div>
+                  <Link to={rollup.website}>Website</Link>
+                </div>
+                <div>
+                  <Link to={rollup.developerDocs}>Developer docs</Link>
+                </div>
+                <div>
+                  <Link to={rollup.l2beat}>L2BEAT</Link>
```
Comment (samajammin) [inline src/components/RollupProductDevDoc.js:53]:
```suggestion
                  <Link to={rollup.l2beat}>Technology and risk summary</Link>
```

I'd recommend a more descriptive link text vs. just "L2BEAT" which most people probably won't know what that means.

Comment (corwintines) [reply src/components/RollupProductDevDoc.js]: I agree, will update with the new merged code, cant commit this suggestion after pulling in @minimalsm suggested changes 😅

**Comment by corwintines** [comment]
@samajammin @minimalsm pulled in your suggested changes, as well as changed the styling a bit to present better in lightmode. I think this is ready for another review now. Thanks again!

File: src/components/RollupProductDevDoc.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                </H4>
+                <ul>
+                  <li>
+                    <Link to={rollup.website}>Website</Link>
+                  </li>
+                  <li>
+                    <Link to={rollup.developerDocs}>Developer docs</Link>
+                  </li>
+                  <li>
+                    <Link to={rollup.l2beat}>Technology and risk summary</Link>
```
Comment (minimalsm) [inline src/components/RollupProductDevDoc.js:53]: We should set these up for translation.

```suggestion
                    <Link to={rollup.website}>Website</Link>
                  </li>
                  <li>
                    <Link to={rollup.developerDocs}>Developer docs</Link>
                  </li>
                  <li>
                    <Link to={rollup.l2beat}>Technology and risk summary</Link>
```

File: src/data/layer-2-data-files/optimistic-rollups.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "developerDocs": "https://developer.offchainlabs.com/docs/developer_quickstart",
+    "l2beat": "https://l2beat.com/projects/arbitrum/",
+    "note": ""
+  },
+  {
+    "name": "Metis Andromeda",
+    "website": "https://metis.io",
+    "developerDocs": "https://docs.metis.io/",
+    "l2beat": "https://l2beat.com/projects/metis/",
+    "note": "Fraud proofs in development"
```
Comment (minimalsm) [inline src/data/layer-2-data-files/optimistic-rollups.json:14]: These notes should be translated too

```suggestion
    "note": "Fraud proofs in development"
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
Should we set this up for translations before merging?

File: src/components/RollupProductDevDoc.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+// Libraries
+import React from "react"
+import styled from "styled-components"
+
+// Components
+import Link from "./Link"
+
+// Data
+import optimisticRollupList from "../data/layer-2-data-files/optimistic-rollups.json"
+import zkRollupList from "../data/layer-2-data-files/zk-rollups.json"
```
Comment (wackerow) [inline src/components/RollupProductDevDoc.js:10]: Discussed at stand-up the idea of combining these into a single data file under `../data/layer-2.json` with root level keys `"optimistic-rollups"` and `"zk-rollups"`.
@corwintines curious your thought on this, if you were partial to keeping them separate for any reason?

File: src/components/RollupProductDevDoc.js (hunk 5 above)
Comment (wackerow) [inline src/components/RollupProductDevDoc.js:53]: On top of translations, a stylistic preference, but destructure here?

```suggestion
      {data[rollupType].map(({ name, note, website, developerDocs, l2beat ) => {
        return (
          <ProductCard>
            <Content>
              <div>
                <H4>
                  {name}
                  {note.length > 0 && ` (${note})`}
                </H4>
                <ul>
                  <li>
                    <Link to={website}>Website</Link>
                  </li>
                  <li>
                    <Link to={developerDocs}>Developer docs</Link>
                  </li>
                  <li>
                    <Link to={l2beat}>Technology and risk summary</Link>
```

File: src/data/layer-2-data-files/optimistic-rollups.json (hunk 6 above)
Comment (wackerow) [reply src/data/layer-2-data-files/optimistic-rollups.json:14]: Yeah agree, we should extract these for translation. I suggested changing the key to `"noteKey"` to remind us this should be a lookup key, and keep the string in the language json.

The name and websites of course don't need to be translated.

**Review by wackerow — CHANGES_REQUESTED** [review]
@corwintines Let me know if you want a hand with any of this, but replied with a couple thoughts just on the organizational side. Let me know your thoughts.

File: src/components/RollupProductDevDoc.js (hunk 5 above)
Comment (corwintines) [reply src/components/RollupProductDevDoc.js:53]: Will get this in now

File: src/data/layer-2-data-files/optimistic-rollups.json (hunk 6 above)
Comment (corwintines) [reply src/data/layer-2-data-files/optimistic-rollups.json:14]: Sounds good, thanks for suggestion. Will get this done in next commit, just wanted to get this in an acceptable place before wrapping that up.

File: src/components/RollupProductDevDoc.js (hunk 7 above)
Comment (corwintines) [reply src/components/RollupProductDevDoc.js:10]: Will combine, definitely open. I don't have a strong opinion for keeping them separate, more just mental map stuff but we can break out if these lists get too large.

**Review by minimalsm — APPROVED** [review]
LGTM! 

IMO we should maybe move these notes out of `/en/common.json/` and into `/en/layer-2.json` when we create the layer 2 page.

**Review by wackerow — APPROVED** [review]
Nice, looking good. Agree with @minimalsm that we should move those strings to another JSON file eventually, but I see the idea behind placing in common since it's a reusable component.

**Comment by corwintines** [comment]
@wackerow definitely will move those into the layer-2.json file when we create that

**Pull request merged**